### PR TITLE
Add Ellipse and Rectangle Schema

### PR DESCRIPTION
**Pull request by hand-dot**
ref: 
- #331
- #330

### CODE REVIEW

File: packages/schemas/src/shapes/rectAndEllipse.ts
Comment (hand-dot) [inline packages/schemas/src/shapes/rectAndEllipse.ts:1]: The main part of this PR is...

**Comment by hand-dot** [comment]
@peteward can you check this PR?

**Comment by hand-dot** [comment]
memo: to fix conflict

https://github.com/pdfme/pdfme/pull/380/files#diff-6686759c7bc6ba95211b21e6b5043a8496afd228513b01a77b4c67834c82fd8b

**Review by peteward — APPROVED** [review]
very cool.

Question - if you want to write a custom plugin that extends one that already exists, is it harder with this change to consolidate the separate files into a single `Plugin` definition?

For example, if I want to have another shape plugin and I want to use the same propPanel as your `shape` one, is it still easy to do this without duplicating the code?

**Comment by hand-dot** [comment]
@peteward 

> Question - if you want to write a custom plugin that extends one that already exists, is it harder with this change to consolidate the separate files into a single Plugin definition?
> For example, if I want to have another shape plugin and I want to use the same propPanel as your shape one, is it still easy to do this without duplicating the code?

we can extend like this.


```ts
const yourShape = {
  ui: () => {},
  pdf: () => {},
  propPanel: {
    ...ellipse.propPanel,
    defaultSchema: {
      ...ellipse.propPanel.defaultSchema,
      type: 'yourShape',
    },
  },
};
```

**Pull request merged**